be/src/exec/olap_scan_prepare.h

```diff
@@ -30,75 +32,149 @@ class ColumnPredicate;
 using ColumnPredicatePtr = std::unique_ptr<ColumnPredicate>;
 using ColumnPredicatePtrs = std::vector<ColumnPredicatePtr>;
 
-class OlapScanConjunctsManager {
-public:
+struct OlapScanConjunctsManagerOptions {
     // fields from olap scan node
-    const std::vector<ExprContext*>* conjunct_ctxs_ptr;
-    const TupleDescriptor* tuple_desc;
-    ObjectPool* obj_pool;
-    const std::vector<std::string>* key_column_names;
-    const RuntimeFilterProbeCollector* runtime_filters;
-    RuntimeState* runtime_state;
+    const std::vector<ExprContext*>* conjunct_ctxs_ptr = nullptr;
+    const TupleDescriptor* tuple_desc = nullptr;
+    ObjectPool* obj_pool = nullptr;
+    const std::vector<std::string>* key_column_names = nullptr;
+    const RuntimeFilterProbeCollector* runtime_filters = nullptr;
+    RuntimeState* runtime_state = nullptr;
+
     int32_t driver_sequence = -1;
 
-private:
-    // fields generated by parsing conjunct ctxs.
-    // same size with |_conjunct_ctxs|, indicate which element has been normalized.
-    std::vector<bool> normalized_conjuncts;
-    std::map<std::string, ColumnValueRangeType> column_value_ranges;  // from conjunct_ctxs
-    OlapScanKeys scan_keys;                                           // from _column_value_ranges
-    std::vector<TCondition> olap_filters;                             // from _column_value_ranges
-    std::vector<TCondition> is_null_vector;                           // from conjunct_ctxs
-    std::map<int, std::vector<ExprContext*>> slot_index_to_expr_ctxs; // from conjunct_ctxs
+    bool scan_keys_unlimited = true;
+    int32_t max_scan_key_num = 1024;
+    bool enable_column_expr_predicate = false;
 
-    // unreached runtime filter and they can push down to storage engine
-    UnarrivedRuntimeFilterList rt_ranger_params;
+    PredicateTreeParams pred_tree_params;
+};
+
+struct RawExprContainer {
+    explicit RawExprContainer(Expr* root_expr);
+
+    Expr* root() const;
+    StatusOr<ExprContext*> expr_context(ObjectPool* obj_pool, RuntimeState* state) const;
+
+    Expr* root_expr;
+    mutable ExprContext* new_expr_ctx = nullptr;
+};
 
+struct ExprContextContainer {
+    explicit ExprContextContainer(ExprContext* expr_ctx);
+
+    Expr* root() const;
+    StatusOr<ExprContext*> expr_context(ObjectPool* obj_pool, RuntimeState* state) const;
+
+    ExprContext* expr_ctx;
+};
```

Comment:
why we need RawExprContainer and ExprContextContainer?
it seems similar to Expr and ExprContext